ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.component.spec.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ Copyright 2017 VMware, Inc. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {ReactiveFormsModule} from '@angular/forms';
+import {ClarityModule} from 'clarity-angular';
+import {HttpModule} from '@angular/http';
+import {CreateVchWizardService} from '../create-vch-wizard.service';
+import {Observable} from 'rxjs/Observable';
+import {NetworksComponent} from './networks.component';
+
+describe('NetworksComponent', () => {
+
+  let component: NetworksComponent;
+  let fixture: ComponentFixture<NetworksComponent>;
+  let service: CreateVchWizardService;
+
+  function setDefaultRequiredValues() {
+    component.form.get('bridgeNetwork').setValue('portGroup');
+    component.form.get('publicNetwork').setValue('portGroup');
+  }
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        ReactiveFormsModule,
+        HttpModule,
+        ClarityModule
+      ],
+      providers: [
+        {
+          provide: CreateVchWizardService,
+          useValue: {
+            getDistributedPortGroups() {
+              return Observable.of([{
+                text: 'portGroup'
+              }]);
+            }
+          }
+        }
+      ],
+      declarations: [
+        NetworksComponent
+      ]
+    });
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(NetworksComponent);
+    component = fixture.componentInstance;
+    component.onPageLoad();
+
+    service = fixture.debugElement.injector.get(CreateVchWizardService);
+    spyOn(service, 'getDistributedPortGroups').and.callThrough();
+  });
+
+  it('should be created', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should start with an invalid form', () => {
+    expect(component.form.invalid).toBe(true);
+  });
+
+  it('should end with an invalid form on step commit without selecting bridge and public network',  () => {
+    component.onCommit();
+    expect(component.form.invalid).toBe(true);
+  });
+
+  it('should end with an valid form on step commit after selecting bridge and public network', () => {
+    setDefaultRequiredValues();
+    component.onCommit();
+    expect(component.form.valid).toBe(true);
+  });
+
+  it('should validate static IP address related fields', () => {
+    const publicNetworkType = component.form.get('publicNetworkType');
```

Comment:
This seems like a good pattern to define these as constants but I don't think this pattern is consistent throughout the spec files. Please double check and make sure we use a consistent pattern.